source/src/plugins/Chi2Test.cc

```diff
@@ -0,0 +1,126 @@
+      public:
+        QTestPtr createQualityTest(const std::string &qname) const override;
+      };
+
+      Chi2Test(const std::string &name);
+      ~Chi2Test() override = default;
+      StatusCode readSettings(const dqm4hep::core::TiXmlHandle xmlHandle) override;
+      void userRun(MonitorElement* monitorElement, QualityTestReport &report) override;
+
+    protected:
+      std::string m_options;
+
+    };
+
+    typedef Chi2Test::Factory Chi2TestFactory;
+
+    //-------------------------------------------------------------------------------------------------
+    //-------------------------------------------------------------------------------------------------
+
+    inline QTestPtr Chi2Test::Factory::createQualityTest(const std::string &qname) const {
+      return std::shared_ptr<QTest>(new Chi2Test(qname));
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    //-------------------------------------------------------------------------------------------------
+
+    Chi2Test::Chi2Test(const std::string &qname)
+        : QualityTest("Chi2", qname),
+	  m_options()
+{
+      m_description = "Performs the chi-squared test on a TH1 and a reference, outputting the p-value. This is analogous to the Kolmogorov-Smirnov "
+	              "test but is appropriate for use on binned data like histograms, which the Kolmogorov test is not.";
+    }
+
+    //-------------------------------------------------------------------------------------------------
+
+    StatusCode Chi2Test::readSettings(const TiXmlHandle xmlHandle) {
+      RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::readParameter(xmlHandle, "Options", m_options));
+
+      return STATUS_CODE_SUCCESS;
+    }
+
+    void Chi2Test::userRun(MonitorElement* pMonitorElement, QualityTestReport &report) {
+      const bool hasObject = (pMonitorElement->object() != nullptr);
+      const bool hasReference = (pMonitorElement->reference() != nullptr);
+      const bool isHistogram = pMonitorElement->object()->InheritsFrom("TH1");
+
+      Option_t* pOptions = &m_options[0u];
+
+      if (!hasObject) {
+        report.m_message = "No object attached to monitor element";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      if (!isHistogram) {
+        report.m_message = "The object must be a TH1";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      if (!hasReference) {
+        report.m_message = "No reference attached to monitor element";
+        report.m_quality = 0.f;
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      TH1* pHistogram = pMonitorElement->objectTo<TH1>();
+      TH1* pReferenceHistogram = pMonitorElement->referenceTo<TH1>();
+      
+      report.m_quality = pHistogram->Chi2Test(pReferenceHistogram, pOptions);
```

Comment:
ROOT documentation says : 

option : 
- "UU" = experiment experiment comparison (unweighted-unweighted) 
- "UW" = experiment MC comparison (unweighted-weighted). Note that the first histogram should be unweighted 
- "WW" = MC MC comparison (weighted-weighted) 
- "NORM" = to be used when one or both of the histograms is scaled but the histogram originally was unweighted 
- by default underflows and overflows are not included: 
   - "OF" = overflows included 
   - "UF" = underflows included   
- "P" = print chi2, ndf, p_value, igood 
- "CHI2" = returns chi2 instead of p-value 
- "CHI2/NDF" = returns χ2/ndf

So it is possible to pass the option "CHI2" using the option parameter you provide, which you don't want. The function will then returns a chi2 instead of a pvalue which is not in [0,1] range. I suggest you implement specific options for each valid ROOT option in this context.
Have a look here for the fit function test : 
- [Options parsing](https://github.com/DQM4HEP/dqm4hep-core/blob/master/source/src/plugins/FitParamInRangeTest.cc#L254)
- [Building the option string](https://github.com/DQM4HEP/dqm4hep-core/blob/master/source/src/plugins/FitParamInRangeTest.cc#L377)

Also, `Option_t` type is actually `char` type, meaning that use can directly pass `m_options.c_str()` in the Chi2Test function.